subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -1540,7 +1540,7 @@ static int light_ctrl_srv_init(struct bt_mesh_model *model)
 		return err;
 	}
 
-	atomic_set_bit(&srv->lightness->flags, LIGHTNESS_SRV_FLAG_EXTENDED_BY_LIGHT_CTRL);
+	atomic_set_bit(&srv->lightness->transient.flags, LIGHTNESS_SRV_FLAG_EXTENDED_BY_LIGHT_CTRL);
```

Comment:
Since flags are stored in emds, they can't be reassigned in the init callback, because they will be rewritten later when `emds_load()` is called. This can happen when, for example, Lightness Server wasn't extended by Light Controller (thus this flags is cleared in emds), and then become extended by Light Controller after DFU. Then, Light Controller will try to set this flag, but it will later be reset by emds. This is a hypothetical case, but from the design point of view, we should not set any state to a non-default value before `emds_load` is called. From what I see, there is no need to store all flags, the only flag that needs to be stored with emds is `is_on`.